src/applications/vaos/referral-appointments/ReviewAndConfirm.jsx

```diff
@@ -0,0 +1,231 @@
+        providerFetchStatus === FETCH_STATUS.succeeded
+      ) {
+        const savedSlot = getSlotById(provider.slots, savedSelectedSlot);
+        if (!savedSlot) {
+          routeToCCPage(history, 'scheduleReferral');
+        }
+        dispatch(setSelectedSlot(savedSlot.id));
+      }
+    },
+    [
+      dispatch,
+      savedSelectedSlot,
+      provider.slots,
+      history,
+      providerFetchStatus,
+      selectedSlot,
+    ],
+  );
+
+  if (loading) {
+    return (
+      <div className="vads-u-margin-y--8" data-testid="loading">
+        <va-loading-indicator message="Loading schedule referral review..." />
+      </div>
+    );
+  }
+
+  if (failed) {
+    return (
+      <va-alert data-testid="error" status="error">
+        <h2>We’re sorry. We’ve run into a problem</h2>
+        <p>
+          We’re having trouble getting your upcoming appointments. Please try
+          again later.
+        </p>
+      </va-alert>
+    );
+  }
+  const headingStyles =
+    'vads-u-margin--0 vads-u-font-family--sans vads-u-font-weight--bold vads-u-font-size--source-sans-normalized';
+  return (
+    <ReferralLayout hasEyebrow>
+      <div>
+        <h1 data-testid="review-heading">Review your appointment details</h1>
+        <hr className="vads-u-margin-y--2" />
+        <div className=" vads-l-grid-container vads-u-padding--0">
+          <div className="vads-l-row">
+            <div className="vads-l-col">
+              <h2 className={headingStyles}>
+                {`${currentReferral.CategoryOfCare} Provider`}
+              </h2>
+            </div>
+          </div>
+        </div>
+        <p className="vads-u-margin--0">
+          {provider.providerName} <br />
+          {provider.orgName}
+        </p>
+        <ProviderAddress
+          address={provider.orgAddress}
+          phone={provider.orgPhone}
+        />
+        <hr className="vads-u-margin-y--2" />
+        <div className=" vads-l-grid-container vads-u-padding--0">
+          <div className="vads-l-row">
+            <div className="vads-l-col">
+              <h2 className={headingStyles}>Date and time</h2>
+            </div>
+            <div className="vads-l-col vads-u-text-align--right">
+              <va-link
+                href={`/my-health/appointments/schedule-referral/date-time?id=${
+                  currentReferral.UUID
+                }`}
+                aria-label="Edit date and time"
+                text="Edit"
+                data-testid="edit-when-information-link"
+                onClick={e => {
+                  e.preventDefault();
+                  routeToPreviousReferralPage(
+                    history,
+                    'reviewAndConfirm',
+                    currentReferral.UUID,
+                  );
+                }}
+              />
+            </div>
+          </div>
+        </div>
+        {slotDetails && (
+          <p className="vads-u-margin--0">
+            <span data-testid="slot-day">
```

Comment:
I agree, I actually do test the text for this in the test.
```    expect(await screen.getByTestId('slot-day')).to.contain.text(
      'Monday, September 9, 2024',
    );```

Reply:
So can we remove the extra spans since and do something link this? `expect(screen.getByText(/Monday, September 9, 2024/i)).to.exist;` You could move the test id up to the <p> and use something like that probably.

Reply:
change pushed!